### PR TITLE
Modified API proposal based on richard's latest comments.

### PR DESCRIPTION
**Pull request by keep94**
This proposal is like the previous except that I get rid of the Sink
type, FlushWriter type, and the Set method on the Logger type.

### CODE REVIEW

File: lib/llog/api.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
+package llog
+
+import (
+	"io"
+)
+
+// Level indicates logging level. The higher the value, the less important.
+// Any non-negative integer is always less important than a named constant.
```
Comment (rgooch) [inline lib/llog/api.go]: I think it's better to have the most important level be the smallest number (0). This is more natural and also makes it easier to implement debug levels.

File: lib/llog/api.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+type Level int
+
+const (
+	// Least important
+	Info Level = -1 - iota
+	Warn
+	// Most important
+	Err
```
Comment (rgooch) [inline lib/llog/api.go]: This is missing Debug and Fatal levels.

File: lib/llog/api.go (hunk 1 above)
Comment (keep94) [inline lib/llog/api.go]: I added Debug, Debugln, Debugf per our discussion.

**Comment by keep94** [comment]
Hi Richard. I have addressed the comments you made. Please have another look. Thanks.

**Comment by rgooch** [comment]
I like this API. Next step is to see if the Go maintainers would accept this API for the standard log package.

**Comment by rgooch** [comment]
Coming back to this proposal now that I have an immediate need for debug logging :-)...
I've put together a simpler API that follows a similar pattern as what I've done with my prefixlogger package, which is to simply wrap an existing Logger to produce a new logger which has Debug*() methods in addition to the standard methods. This also makes the implementation trivial.

I've uploaded the implementation to my repository and a little later I'll merge it upstream. Here is it: https://github.com/rgooch/Dominator/blob/master/lib/log/debuglogger/api.go

Please take a look and comment.

**Comment by rgooch** [comment]
We have the lib/log/debuglogger package which is a minimalist solution and should be sufficient.

**Pull request closed without merging**